### DISCUSSION TITLE
Informational response not treated as a response for `httpcore.stream`.

### DISCUSSION THREAD

**Discussion by A5rocks** [encode/httpcore#572] [Potential Issue] [answered]
A websocket handshake requires the client send a certain request, and the server respond with a 101 status code (indicating that both should switch to websockets protocol). However, HTTPCore continues a loop for an event until it gets an `h11.Response` ... NOT a `h11.InformationalResponse`. Thus, the call hangs forever. (Or until some timeout, I don't have the patience to see.)

HTTPCore should probably support this usecase.

Reproducer:
```
import httpcore


with httpcore.stream("GET", "https://httpbin.org/status/101", headers={b"Upgrade": b"WebSocket"}):
    pass
```

I was under the impression this was the intended method for websockets due to https://github.com/encode/httpx/issues/304#issuecomment-1046894140.

**Comment by lovelydinosaur** [comment, marked as answer]
> I was under the impression

I think you've got the right impression here,  yes, but haven't had a chance to look at it yet. Will try to do so later this week, but nudge me if I don't get there...

**Reply by lovelydinosaur** [reply]
Okay, now resolved here... https://github.com/encode/httpcore/pull/581
